apps/console/src/features/identity-providers/components/wizards/hypr/hypr-authentication-provider-create-wizard-content.tsx

```diff
@@ -0,0 +1,325 @@
+     */
+    triggerPrevious: (previousFunctionCb: () => void) => void;
+    /**
+     * Callback to change the wizard page,
+     * @param pageNo - Page Number.
+     */
+    changePageNumber: (pageNo: number) => void;
+    /**
+     * IDP template.
+     */
+    template: IdentityProviderTemplateInterface;
+    /**
+     * Total wizard page count.
+     * @param pageCount - Page number.
+     */
+    setTotalPage: (pageCount: number) => void;
+    /**
+     * Callback to be triggered for form submit.
+     * @param values - Form values.
+     */
+    onSubmit: (values: HyprAuthenticationProviderCreateWizardFormValuesInterface) => void;
+}
+
+const FORM_ID: string = "hypr-authenticator-wizard-form";
+
+/**
+ * Hypr Authentication Provider Create Wizard content component.
+ *
+ * @param props - Props injected to the component.
+ * @returns Functional component.
+ */
+export const HyprAuthenticationProviderCreateWizardContent: FunctionComponent<
+    HyprAuthenticationProviderCreateWizardContentPropsInterface
+> = (
+    props: HyprAuthenticationProviderCreateWizardContentPropsInterface
+): ReactElement => {
+
+    const {
+        triggerSubmission,
+        triggerPrevious,
+        changePageNumber,
+        template,
+        setTotalPage,
+        onSubmit,
+        [ "data-componentid" ]: componentId
+    } = props;
+
+    const { t } = useTranslation();
+
+    const [ idpList, setIdPList ] = useState<IdentityProviderListResponseInterface>({});
+    const [ isIdPListRequestLoading, setIdPListRequestLoading ] = useState<boolean>(undefined);
+
+    /**
+     * Loads the identity provider authenticators on initial component load.
+     */
+    useEffect(() => {
+
+        getIDPlist();
+    }, []);
+
+    /**
+     * Get Idp List.
+     */
+    const getIDPlist = (): void => {
+
+        setIdPListRequestLoading(true);
+
+        getIdentityProviderList(null, null, null)
+            .then((response: IdentityProviderListResponseInterface) => {
+                setIdPList(response);
+            })
+            .catch((error: Promise<void>) => {
+                handleGetIDPListCallError(error);
+            })
+            .finally(() => {
+                setIdPListRequestLoading(false);
+            });
+    };
+
+    /**
+     * Check whether IDP name is already exist or not.
+     *
+     * @param value - IDP name.
+     * @returns error msg if name is already taken.
+     */
+    const idpNameValidation = (value: string): string => {
+
+        let nameExist: boolean = false;
+
+        if (idpList?.count > 0) {
+            idpList?.identityProviders.map((idp: IdentityProviderTemplateInterface) => {
+                if (idp?.name === value) {
+                    nameExist = true;
+                }
+            });
+        }
+        if (nameExist) {
+            return t("console:develop.features." +
+                "authenticationProvider.forms.generalDetails.name." +
+                "validations.duplicate");
+        }
```

Comment:
Can we simplify this by moving the return statement (line 140) to inside the if blocks (line 135) and removing the local variable nameExist?

Reply:
addressed with a new PR